File: quiz/students.py

import cs50

class Student:
    def __init__(self, name, dorm):
        self.name = name
        self.dorm = dorm


def main():
    students = [
        Student("Stelios", "Branford"),
        Student("Maria", "Cabot"),
        Student("Anushree", "Ezra Stiles"),
        Student("Brian", "Winthrop")
    ]
    print("Before:")
    for student in students:
        print(f"{student.name} from {student.dorm}")
    students = sorted(students, key= lambda student: student.name)
    print("After:")

    for student in students:
        print(f"{student.name} from {student.dorm}")


if __name__ == "__main__":
    main()

File: quiz/students.js
let students = [
    {name: "Stelios", dorm: "Branford"},
    {name: "Maria", dorm: "Cabot"},
    {name: "Anushree", dorm: "Ezra Stiles"},
    {name: "Brian", dorm: "Winthrop"}
];

console.log("Before:")
for (let student of students)
{
    console.log(student.name + " from " + student.dorm);
}
students.sort(cmp);
console.log("After:")
for (let student of students)
{
    console.log(student.name + " from " + student.dorm);
}

function cmp(a, b)
{
    order = a.name.localeCompare(b.name)
    return(order)
}
